src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransfer.cs

```diff
@@ -194,6 +194,21 @@ public static bool TemplatesMatch(Network network, Transaction partialTransactio
             return true;
         }
 
+        /// <inheritdoc />
+        public int GetSignatureCount()
+        {
+            Guard.NotNull(this.PartialTransaction, nameof(this.PartialTransaction));
+            Guard.Assert(this.PartialTransaction.Inputs.Any());
+
+            Script scriptSig = this.PartialTransaction.Inputs[0].ScriptSig;
+            if (scriptSig == null)
+                return 0;
+
+            Op[] ops = scriptSig.ToOps().ToArray();
+
+            return ops.Length - (1 + ops.Count(o => o.Code == OpcodeType.OP_0));
```

Comment:
This seems a bit hacky, doesn't the P2SH template have a helper method for extracting the parameters? Or does that not work with a partially signed transaction?

Reply:
can use
`TransactionSignature[] result = PayToMultiSigTemplate.Instance.ExtractScriptSigParameters(this.network, new Script(scriptSig));`